rust/windows-client/src-tauri/src/client.rs

```diff
@@ -0,0 +1,69 @@
+use anyhow::Result;
+use clap::Parser;
+use cli::CliCommands as Cmd;
+
+mod cli;
+mod debug_commands;
+mod device_id;
+#[cfg(target_family = "unix")]
+mod gui {
+    use super::*;
+
+    pub enum ControllerRequest {
+        SignIn,
+    }
+
+    pub(crate) fn run(_: GuiParams) -> Result<()> {
+        // The Ubuntu CI runner doesn't have gdk and some other Tauri deps installed, so it fails unless we stub out the GUI.
+        panic!("The Tauri GUI isn't implemented for Linux.");
+    }
+}
+#[cfg(target_os = "windows")]
+mod gui;
+mod local_webserver;
+// Relies on some types from Tauri
+#[cfg(target_os = "windows")]
+mod settings;
+
+/// Prevents a problem where changing the args to `gui::run` breaks static analysis on non-Windows targets, where the gui is stubbed out
+#[allow(dead_code)]
+pub(crate) struct GuiParams {
+    /// The URL of an incoming deep link from a web browser
+    deep_link: Option<String>,
+    /// True if we should slow down I/O operations to test how the GUI handles slow I/O
+    inject_faults: bool,
```

Comment:
I think we shouldn't have a variable like this, instead we can have different binaries for this cases, that we generate with generic/traits that add this feature for debugging.

For now we can just add a TODO and remove this later before launching the windows client.

Reply:
@ReactorScram Yeah was chatting with Gabi about these. We converged on leaving these debug helpers in to help with debugging and quick iteration/learning, but with a `TODO` to remove or refactor later so that we minimize debug logic in the release paths.

Reply:
@conectado What if they're behind a Cargo feature flag or something so they only build in debug mode?